app/src/main/java/com/humara/nagar/ui/report/ReportFragment.kt

```diff
@@ -23,23 +43,216 @@ class ReportFragment : BaseFragment() {
+
+    private fun takePhotoFromCamera() {
+        val permissions = arrayOf(
+            Manifest.permission.CAMERA
+        )
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else {
+            //Take picture
+        }
+    }
+
+
+    @Deprecated("Deprecated in Java")
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        var allPermissionsGranted = true
+        val deniedPermissions = mutableListOf<String>()
+        for (i in requestedPermissions.indices) {
+            if (grantResults[i] != PackageManager.PERMISSION_GRANTED) {
+                Logger.debugLog("NOT GRANTED ${requestedPermissions[i]}")
+                allPermissionsGranted = false
+                deniedPermissions.add(requestedPermissions[i])
+                break
+            } else {
+                Logger.debugLog("GRANTED ${requestedPermissions[i]}")
+                break
+            }
+        }
+        if (allPermissionsGranted) {
+            // All permissions were granted. You can proceed with the action that requires these permissions.
+            if (permissions.contains(Manifest.permission.ACCESS_FINE_LOCATION) &&
+                    permissions.contains(Manifest.permission.ACCESS_COARSE_LOCATION)) {
+                getAddress()
+            }
+        } else {
+            // One or more permissions were denied. You should show a message to the user explaining why you need these permissions.
+            Logger.debugLog("Denied permission: $deniedPermissions")
+            handlePermissionDenied(deniedPermissions.toTypedArray())
+        }
+
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    }
+
+    fun handlePermissionDenied(permissions: Array<String>) {
+        val showPermissionRationalePopup: Boolean = permissions.isNotEmpty()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && showPermissionRationalePopup) {
+            for (permission in permissions) {
+                showRationaleDialog(
+                    resources.getString(R.string.permission_required_title),
+                resources.getString(R.string.premission_required_description),
+                permission, PermissionsUtils.REQUEST_CODE
+                )
+                break
+            }
+        } else {
+            gotoAppSettingsPermission()
+        }
+    }
+
+    fun showRationaleDialog(title: String, message: String, permission: String, requestCode: Int) {
+        val builder: AlertDialog.Builder = AlertDialog.Builder(requireContext())
+        builder.setTitle(title)
+            .setMessage(message)
+            .setPositiveButton(resources.getString(R.string.ok)) { _, _ ->
```

Comment:
no need of Ok button here

Reply:
Okay, got it.

Reply:
Fixed. Added gotoSettings() in place of positiveButton onClick.